src/DynamoDbSession.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace pixelandtonic\dynamodb;
+
+use Aws\DynamoDb\Exception\DynamoDbException;
+use Yii;
+use yii\base\InvalidConfigException;
+use yii\di\Instance;
+use yii\web\Session;
+
+class DynamoDbSession extends Session
+{
+    public DynamoDbConnection|string|array $dynamoDb = 'dynamoDb';
+    public string $dataAttribute = 'data';
+
+    /**
+     * @var bool Whether to allow garbage collection. Most often this should be false,
+     *           and DynamoDB's ttl settings should be used.
+     */
+    public bool $allowGc = false;
+
+    /**
+     * @inheritDoc
+     * @throws InvalidConfigException
+     */
+    public function init(): void
+    {
+        parent::init();
+        $this->dynamoDb = Instance::ensure($this->dynamoDb, DynamoDbConnection::class);
+        $this->setTimeout($this->getTimeout());
+    }
+
+    public function setTimeout($value): void
+    {
+        parent::setTimeout($value);
+
+        // Prevent premature set from constructor
+        if ($this->dynamoDb instanceof DynamoDbConnection) {
+            $this->dynamoDb->ttl = $value;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getUseCustomStorage(): bool
+    {
+        return true;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function readSession($id): string
+    {
+        try {
+            $item = $this->dynamoDb->getItem($id);
+        } catch (DynamoDbException $e) {
+            Yii::error("Unable to read session: {$e->getMessage()}", __METHOD__);
+            return '';
+        }
+
+        $data = $item[$this->dataAttribute] ?? '';
```

Comment:
This seems kind of suspect, shouldn't those be set and if the item returned does not have those it probably was not a session item?